### ISSUE TITLE
Husky throws an ArgumentException when I try to commit through Sourcetree

### ISSUE THREAD

**Issue by AntonisManiatis** [alirezanet/Husky.Net#96]
### Version

0.6.2

### Details

I am not so sure if this should be opened as a bug or a discussion would be a better fit so apologies in advance but basically
whenever I try to commit changes through [Sourcetree](https://www.sourcetreeapp.com/) Husky throws an ArgumentException.

![husky-exception](https://github.com/alirezanet/Husky.Net/assets/16625782/eda3e1de-680b-4832-b3d9-96ae0e0078c6)

Here's the stack trace:
```
Unhandled exception. System.ArgumentException: An item with the same key has already been added. Key: HOME
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at CliFx.Utils.Extensions.CollectionExtensions.ToDictionary[TKey,TValue](IDictionary dictionary, IEqualityComparer`1 comparer) in /_/CliFx/Utils/Extensions/CollectionExtensions.cs:line 38
   at CliFx.CliApplication.RunAsync(IReadOnlyList`1 commandLineArguments) in /_/CliFx/CliApplication.cs:line 216
   at Program.<Main>$(String[] args) in /home/runner/work/Husky.Net/Husky.Net/src/Husky/Program.cs:line 31
   at Program.<Main>(String[] args)
```

It works as expected if I try to commit either from the CLI, VS or VSCode but not on Sourcetree. I had a very quick look at the source code to try and figure out what could be happening but couldn't figure anything out immediately. Any idea what that Key: HOME might be or where it comes from? is it an environment variable? I'd appreciate any feedback on this, we'd really like to use your tool but without having to sacrificing Sourcetree in the process 😄. Thanks!

OS: Windows 11 Pro 22621.2428
Sourcetree version: 3.4.15

### Steps to reproduce

- Install & setup husky (as instructed by the [Getting Started](https://alirezanet.github.io/Husky.Net/guide/getting-started.html) page) on a .NET project with a Task Runner (I copied the real-world example found in [task.runner.json].(https://alirezanet.github.io/Husky.Net/guide/task-runner.html#task-runner-json) but only kept the dotnet-format command) 
- Download and install [Sourcetree](https://www.sourcetreeapp.com/) open the project's repository.
- Make a change to any file in the repo and try to commit from Sourcetree.

**Comment by alirezanet** [comment]
Hi @AntonisManiatis,
This is a bug in Sourcetree, you should've opened this issue on their repo, however, I did a small research and found this issue:
https://community.atlassian.com/t5/Sourcetree-questions/Sourcetree-2-6-10-ArgumentException-encountered-An-item-with-the/qaq-p/909718

check it out, it might also solve your problem.

**Comment by alirezanet** [comment]
I might be wrong let me try to reproduce it, I just noticed the exception is raised from Husky!

**Comment by alirezanet** [comment]
Hi again @AntonisManiatis,
I think for some reason sourcetree is sending additional arguments to the hooks! 
can you share your git and credential manager versions?

![image](https://github.com/alirezanet/Husky.Net/assets/7004080/f97865bd-8092-4c09-a7f9-c331df5a95a7)

**Comment by AntonisManiatis** [comment]
Hi @alirezanet,

Here's my git & git credential-manager versions:

![image](https://github.com/alirezanet/Husky.Net/assets/16625782/67860c96-f940-48dd-b2de-b8e2035b359b)

I checked my environment variables and nothing seems out of the ordinary, no duplicates etc. Also a colleague of mine tried on his machine (again windows 11 & latest Sourcetree) and it's the same story.

I wrote a little program to print command line arguments and environment variables to standard out, changed the pre-commit hook to call that program instead of doing `dotnet husky run --group pre-commit` , commited a change through Sourcetree and I didn't find any duplicate env variables printed.

**Comment by alirezanet** [comment]
Hi @AntonisManiatis,
I found the problem, for some reason, Sourcetree is adding a lower-case `home` environment variable and that is causing the issue, I'm working on a fix for this problem ...
if you want to also test it you can change your `pre-commit` hook to log the environment variables this way if you want:

``` bash
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

# Log all environment variables
printenv

dotnet husky run -v --group "pre-commit"
```

**Comment by alirezanet** [comment]
Hi @AntonisManiatis,
This is fixed in `v0.6.3`.

**Comment by AntonisManiatis** [comment]
thank you very much @alirezanet, cheers!
